### PR TITLE
Shell add help

### PR DESCRIPTION
**Pull request by ChristopherMSpencer**
Added CratonShell class with basic parsing for -h, --help args
Added test cases:
- Exception Handling
- Help args (-h, --help)
- No args
- Modified original basic test case to verify happy path with mocked CratonShell

### CODE REVIEW

File: cratonclient/tests/base.py [hunk 1]
```diff
@@ -15,9 +15,31 @@
...
+import sys
+import six
 from oslotest import base
 
+from cratonclient.shell import main
 
 class TestCase(base.BaseTestCase):
 
     """Test case base class for all unit tests."""
+    def shell(self,arg_str, exitcodes=(0,)):
```
Comment (ChristopherMSpencer) [inline cratonclient/tests/base.py]: Does this belong here or should I move it to tests.shell.base or something similar?

File: cratonclient/shell/main.py [hunk 2]
```diff
@@ -14,8 +14,45 @@
...
+        parser = self.get_base_parser()
+        (options, args) = parser.parse_known_args(argv)
+        if options.help or not argv:
+            parser.print_help()
+            return 0
+
 
 def main():
+    try:
+        CratonShell().main(map(encodeutils.safe_decode,sys.argv[1:]))
```
Comment (sigmavirus24) [inline cratonclient/shell/main.py]: Did you run `tox -e pep8`?

File: cratonclient/tests/base.py (hunk 1 above)
Comment (sigmavirus24) [inline cratonclient/tests/base.py]: You should have a base test case for all the shell tests, so no this does not belong here.

Comment (ChristopherMSpencer) [inline cratonclient/tests/base.py]: is it common practice to create a new file for that or just create a new class within this file?

**Comment by sigmavirus24** [comment]
@ChristopherMSpencer can you fix the merge conflicts by rebasing against master?

File: cratonclient/tests/base.py [hunk 3]
```diff
@@ -16,8 +16,38 @@
...
+
+class ShellTestCase(base.BaseTestCase):
+    """Test case base class for all shell unit tests."""
+
+    def shell(self, arg_str, exitcodes=(0,)):
+        """Main function for exercising the craton shell."""
+        orig_stdout = sys.stdout
+        orig_stderr = sys.stderr
+        try:
+            sys.stdout = six.StringIO()
```
Comment (sigmavirus24) [inline cratonclient/tests/base.py]: We should be using mock.patch here, not manually patching things out.

Comment (ChristopherMSpencer) [inline cratonclient/tests/base.py]: made the update.  Is that what you had in mind?

File: cratonclient/tests/base.py [hunk 4]
```diff
@@ -35,12 +35,12 @@ class ShellTestCase(base.BaseTestCase):
     def shell(self, arg_str, exitcodes=(0,)):
         """Main function for exercising the craton shell."""
         with mock.patch('sys.stdout', new=six.StringIO()) as mock_stdout, \
-             mock.patch('sys.stderr', new=six.StringIO()) as mock_stderr:
+                mock.patch('sys.stderr', new=six.StringIO()) as mock_stderr:
```
Comment (sigmavirus24) [inline cratonclient/tests/base.py]: This is overly indented.

Comment (ChristopherMSpencer) [inline cratonclient/tests/base.py]: sorry about that.  Not sure why it was so far indented.

**Pull request merged**